cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -201,6 +210,172 @@ public void getArtifactInfo(HttpRequest request, HttpResponder responder,
   }
 
   @GET
+  @Path("/namespaces/{namespace-id}/artifacts/{artifact-name}/versions/{artifact-version}/properties")
+  public void getProperties(HttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId,
+                            @PathParam("artifact-name") String artifactName,
+                            @PathParam("artifact-version") String artifactVersion,
+                            @QueryParam("scope") @DefaultValue("user") String scope,
+                            @QueryParam("keys") @Nullable String keys)
+    throws NamespaceNotFoundException, BadRequestException {
+
+    Id.Namespace namespace = validateAndGetNamespace(namespaceId, scope);
+    Id.Artifact artifactId = validateAndGetArtifactId(namespace, artifactName, artifactVersion);
+
+    try {
+      ArtifactDetail artifactDetail = artifactRepository.getArtifact(artifactId);
+      Map<String, String> properties = artifactDetail.getMeta().getProperties();
+      Map<String, String> result;
+
+      if (keys != null && !keys.isEmpty()) {
+        result = new HashMap<>();
+        for (String key : Splitter.on(',').trimResults().split(keys)) {
+          result.put(key, properties.get(key));
+        }
+      } else {
+        result = properties;
+      }
+      responder.sendJson(HttpResponseStatus.OK, result);
+    } catch (ArtifactNotFoundException e) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, "Artifact " + artifactId + " not found.");
+    } catch (IOException e) {
+      LOG.error("Exception reading artifacts named {} for namespace {} from the store.", artifactName, namespaceId, e);
+      responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR,
+                           "Error reading artifact properties from the store.");
+    }
+  }
+
+  @PUT
+  @Path("/namespaces/{namespace-id}/artifacts/{artifact-name}/versions/{artifact-version}/properties")
+  public void writeProperties(HttpRequest request, HttpResponder responder,
+                              @PathParam("namespace-id") String namespaceId,
+                              @PathParam("artifact-name") String artifactName,
+                              @PathParam("artifact-version") String artifactVersion)
```

Comment:
don't think it makes sense to be able to write system properties from another namespace, so it is read-only